### PR TITLE
Object code export import additions fix

### PR DESCRIPTION
**Pull request by kjmcqueen**
## Overview
Cleanup, Additions and refactor Object Code Export/Import

- Import now subtitues for boot.rbl load
- Cleaned up Ob.proto by removing unused messages and renumbering fields
-- This will require previously exported code to be re-exported
- Added full support for ExtendedLocation object in export
- Corrected opcode export to use Instr (16-bit words) rather than uint32_t for size calculation
- ObjectInfo struct is now a class to allow for a constructor
- Import creates objects recursively by Code litvec. This is more similar to the way export is done to more closely mimic the just-in-time compile/execute used by rosette when loading boot.rbl
- Handle object references during import
- Remove some unused object that were causing import difficulties
- Support ExpandedLocation import
- Fully create objects rather than create with NULL pointers and try to fill in later.
- 

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/ROS-425
https://rchain.atlassian.net/browse/ROS-437

### Complete this checklist before you submit the PR
- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Sorry for the huge PR. I've split it up into separate commits to help with reviewing.

### CODE REVIEW

**Comment by pyrocto** [comment]
Please give a more thorough commit message.

**Comment by kjmcqueen** [comment]
I added more detail in the description above PTAL.

File: rosette/src/Import.cc [hunk 1]
```diff
@@ -24,150 +24,359 @@ extern pOb NILmeta;
...
-        break;
-    case ObjectCodePB::OT_TUPLE_EXPR:
-        retval = TupleExpr::create();
+    }
+
+    case ObjectCodePB::OT_TEMPLATE: {
+        ObjectCodePB::Template * pbob = ob->mutable_template_();
+        retval = Template::create( (Tuple *)createRosetteObject(pbob->mutable_keytuple()),
+                                    createRosetteObject(pbob->mutable_pat()),
+                                    (CompoundPattern *)createRosetteObject(pbob->mutable_keymeta()) );
```
Comment (tymm) [inline rosette/src/Import.cc]: Why do you cast `keymeta` to `CompoundPattern*`?

Comment (kjmcqueen) [reply rosette/src/Import.cc]: Because I inadvertently had the keymeta and pat parameters swapped in the call to Template::create(). I've fixed it.

**Pull request merged**